### PR TITLE
Reviews and Rating (VTEX)

### PR DESCRIPTION
**Pull request by gabekerchner**
## What is this contribution about?
Brings the Rating and Review from context of product (VTEX)

https://www.loom.com/share/79670af8396046809a7a165fb075f8ff?sid=6dabb9e3-3357-4524-abf5-315fff7e1f82

image
image

https://deco-sites-interdental--feat-reviews-and-rating.deno.dev/produto-teste-1/p?skuId=32213

### CODE REVIEW

File: scripts/start.ts [hunk 1]
```diff
@@ -42,10 +42,10 @@ function processTypeInNestedObject(obj: any) {
 
 for await (const entry of walk(".")) {
   if (entry.isFile) {
-    if (entry.path.endsWith(OPENAPI_EXTENSION)) {
+    if (entry.path.endsWith(OPENAPI_EXTENSION) && (entry.path === "vtex\\utils\\openapi\\my.openapi.json")) {
```
Comment (guitavano) [inline scripts/start.ts:45]: Should be here?

File: vtex/loaders/product/extend.ts [hunk 2]
```diff
@@ -87,13 +88,68 @@ const variantsExt = async (
...
+        reviewHeadline: productReviews[reviewIndex]?.title,
+        reviewBody: productReviews[reviewIndex]?.text,
+        reviewRating: {
+          "@type": "AggregateRating",
+          ratingValue: productReviews[reviewIndex]?.rating || 0
+        }
+      }))
+    };
+  });
+};
```
Comment (guitavano) [inline vtex/loaders/product/extend.ts:143]: Can you put it inside a toReview() function in transform.ts file?

File: vtex/loaders/workflow/product.ts [hunk 3]
```diff
@@ -55,6 +56,17 @@ const loader = async (
...
+    ["GET /reviews-and-ratings/api/rating/:productId"]({
+      productId: props.productID,
+    }).then((res) => res.json());
+
+
+    const productReviews: any = await my
+    ["GET /reviews-and-ratings/api/reviews"]({
+      productId: props.productID,
+    }).then((res) => res.json());
+
```
Comment (guitavano) [inline vtex/loaders/workflow/product.ts:69]: Should be here?

Comment (gabekerchner) [reply vtex/loaders/workflow/product.ts]: Fixed

File: scripts/start.ts (hunk 1 above)
Comment (gabekerchner) [reply scripts/start.ts]: I don't know why but path doesn't recognize a new entry and doesn't generate a gen.api

**Review by guitavano — APPROVED** [review]

**Pull request merged**